File: lib_rb/human.rb
require 'game'

class Human
  def initialize
    @game = Game.new
    @moves = {
      "\e[A\n" => 'up',
      "\e[C\n" => 'right',
      "\e[D\n" => 'left',
      "\e[B\n" => 'down'
    }
    @moves.default = 'invalid move'
    display
  end

  def display
    puts @game.board[0...4].join(' ')
    puts @game.board[4...8].join(' ')
    puts @game.board[8...12].join(' ')
    puts @game.board[12...16].join(' ')
  end

  def move(direction)
    @game.move @moves[direction]
    # system "clear" or system "cls"
    display
  end

  def play 
    direction = ''
    while !(direction.strip == "q" || direction.strip == "quit" || direction.strip == 'exit' || @game.complete?) do
      direction = gets
      move direction
    end

    @game.highest_tile
  end
end

File: lib_py/human.py
from game import Game, grouper

class Human:
  def __init__(self):
    self.game = Game()
    self.moves = {
      "\x1b[A": 'up',
      "\x1b[C": 'right',
      "\x1b[D": 'left',
      "\x1b[B": 'down'
    }
    self.display()

  def play(self):
    print('playing')
    direction_trimmed = ''
    while not (direction_trimmed == "q" or direction_trimmed == "quit" or direction_trimmed == 'exit' or self.game.is_complete()):
      direction = input()
      direction_trimmed = direction.replace(r"\s", "")
      self.move(direction)

    return (self.game.highest_tile(), self.game.score)

  def display(self):
    for row in grouper(self.game.board, 4):
      print(' '.join(map(str, row)))

  def move(self, direction):
    human_readable_move = self.get_direction(direction)
    self.game.move(human_readable_move)
    print(human_readable_move)
    self.display()

  def get_direction(self, direction):
    try:
      return self.moves[direction]
    except:
      return 'invalid move'

